### ISSUE TITLE
math behind continuous loss function

### ISSUE THREAD

**Issue by nyck33** [OctThe16th/PPO-Keras#6]
```
def proximal_policy_optimization_loss_continuous(advantage, old_prediction):
    def loss(y_true, y_pred):
        var = K.square(NOISE)
        pi = 3.1415926
        denom = K.sqrt(2 * pi * var)
        prob_num = K.exp(- K.square(y_true - y_pred) / (2 * var))
        old_prob_num = K.exp(- K.square(y_true - old_prediction) / (2 * var))

        prob = prob_num/denom
        old_prob = old_prob_num/denom
        r = prob/(old_prob + 1e-10)

        return -K.mean(K.minimum(r * advantage, K.clip(r, min_value=1 - LOSS_CLIPPING, max_value=1 + LOSS_CLIPPING) * advantage))
    return loss

```

I read something about having to get the probability density of a Gaussian output from a neural network for PPO but I still have no clue how to implement this.  You don't seem to be following any of this but can you tell me the math behind the calculation of probability and or papers that showed you how to do it this way?

**Comment by troyscribner** [comment]
If you start with equation 6 from the original paper, https://arxiv.org/pdf/1707.06347.pdf, you see the probability ratio rt(θ) is the new policy divided by the old policy: rt(θ)  =πθ(at|st) / πθold(at|st)

In your example, that's this:
r = prob/(old_prob + 1e-10)

I think the new and old policies are just the probability density function for the normal distribution.

https://en.wikipedia.org/wiki/Normal_distribution

**Comment by nyck33** [comment]
@troyscribner 
Thank you for pointing that out.  I am studying probability recently so it makes more sense.
